qiime2_TagSequencingReport.Rmd

```diff
@@ -0,0 +1,383 @@
+###   read in data
+### ===========================
+
+### read mapping file
+s <- read_qiime_mapping_file(mapping_file_fp) 
+
+### Check if the sample names start with a number. If they do, add an "X" to the front
+toFix = grepl("^[0-9]", s$SampleID)
+s$SampleIDorig = s$SampleID
+s$SampleID[toFix] = paste0('X', s$SampleID[toFix])
+
+### read otu table
+o <- read.table(feature_table_fp, skip=1, header=T, as.is=T, sep="\t", comment.char="", fill=T)
+rownames(o) <- o$X.OTU.ID
+counts <- o %>%
+  select(-X.OTU.ID, -taxonomy)
+
+### taxonomic assignment
+ta <- read.table(file=taxo_assignment_fp, header=T, sep="\t", as.is=T)
+ta$trunc_taxon <- sub("(; [kpcofgs]__)+$", "", ta$Taxon, perl=T)
+rownames(ta) <- ta$Feature.ID
+ta <- ta[rownames(o), c("trunc_taxon", "Confidence")]
+md <- ta$trunc_taxon
+names(md) <- rownames(ta)
+adf <- split_assignments(md) 
+
+### remove contamination
+is_mitochondrial <- grepl("mitochondria", adf$Family)
+is_chloroplast <- grepl("Chloroplast", adf$Class)
+is_unassigned <- grepl("Unassigned", adf$Kingdom)
+is_archaea <- grepl("Archaea", adf$Kingdom)
+is_contam <- is_mitochondrial | is_chloroplast | is_unassigned
+counts <- counts[!is_contam,]
+adf <- adf[!is_contam,]
+rm(is_contam, is_mitochondrial, is_chloroplast, is_unassigned, is_archaea)
+
+### read counts
+read_counts <- read.table(file=read_count_fp, header=F, as.is=T, sep=",") %>%
+  dplyr::rename(SampleIDorig=V1) %>%
+  dplyr::rename(Read_Counts=V2)
+s <- merge(s, read_counts, by="SampleIDorig", all.x=T)
+
+### number of observed OTUs
+notu <- read.table(file=notu_fp, header=T, sep="\t", as.is=T) %>%
+  dplyr::rename(SampleIDorig=X)
+s <- merge(s, notu, by="SampleIDorig", all.x=T)
+
+### Shannon measure
+shannon <- read.table(file=shannon_fp, header=T, sep="\t", as.is=T) %>%
+  dplyr::rename(SampleIDorig=X)
+s <- merge(s, shannon, by="SampleIDorig", all.x=T)
+```
+
+```{r}
+### ===========================
+###   data preprocessing and etc
+### ===========================
+
+### possible issue 1: any mismatch between s$SampleID and colames(counts)?
+#in_s_not_in_counts <- setdiff(s$SampleID, colnames(counts))
+#if (length(in_s_not_in_counts) > 0) stop (simpleError("These SampleID(s) are in the mapping file, but not found in the feature table.", paste(in_s_not_in_counts, collapse=" ")))
+#in_counts_not_in_s <- setdiff(colnames(counts), s$SampleID)
+#if (length(in_counts_not_in_s) > 0) stop (simpleError("These SampleID(s) are in the feature table, but not found in the mapping file.", paste(in_counts_not_in_s, collapse=" ")))
+common_sample_list <- intersect(s$SampleID, colnames(counts))
+s <- s[s$SampleID %in% common_sample_list, ]
+counts <- counts[, common_sample_list]
+
+### possible issue 2: keep only samples that are related with the investigator
+s <- s[s$investigator=="Mark Brown", ]
+counts <- counts[, s$SampleID]
+
+### misc data preprocessing
+color_by <- "time_collected"
+shape_by <- "cage_number"
+s$cage_number <- as.factor(s$cage_number)
```

Comment:
Probably not needed.  We should develop a policy for the caging column so that it starts with a letter.  If the conversion is needed, we should move this up so the conversion happens just after the sample table is imported.

Reply:
deleted